### PR TITLE
Rewrite: More options, better documentation

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by Attamusc** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -1,19 +1,81 @@
...
+Default: `'local'`
+
+Webdriver location. Options are `local` or `remote`. *Note:* Must be given a webdriver url in `webdriver.url` if this option is set to `remote`.
+
+#### webdriver
+Type: `object`
+
+Object containing webdriver options, which include:
+
+##### browser
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 2]
```diff
@@ -1,19 +1,81 @@
...
+Type: `string`  
+Default: `'firefox'`
+
+Desired browser driver to run tests. *Note:* Drivers other than firefox must be downloaded and installed individually.
+
+##### url
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 3]
```diff
@@ -1,19 +1,81 @@
...
+Type: `string`
+
+Remote webdriver's URL. Only used if `target` is set to `remote`.
+
+#### mochaOptions
+Type: `object`
+
+Object containing mocha options, which include:
+
+##### bail
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 4]
```diff
@@ -1,19 +1,81 @@
...
+Type: `boolean`  
+Default: `false`
+
+If `true`, stops test after first failure
+
+##### grep
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 5]
```diff
@@ -1,19 +1,81 @@
...
+
+Only run tests with descriptions that match the pattern given  
+Example: Passing `'Does this thing'` will only run tests that have "Does this thing" in their description
+
+#### options
+Type: `object`
+
+Object containing additional configuration options, which include:
+
+##### parallel
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 6]
```diff
@@ -1,19 +1,81 @@
...
+Type: `boolean`  
+Default: `false`
+
+Boolean representing whether tests should be run in parallel or Note
+
+##### paigeOptions
```
Comment (mrjoelkemp) [inline README.md]: Should this be 5 pound signs or 4?

File: README.md [hunk 7]
```diff
@@ -1,19 +1,81 @@
...
+#### options
+Type: `object`
+
+Object containing additional configuration options, which include:
+
+##### parallel
+Type: `boolean`  
+Default: `false`
+
+Boolean representing whether tests should be run in parallel or Note
```
Comment (mrjoelkemp) [inline README.md]: `or not`

File: README.md (hunk 5 above)
Comment (mrjoelkemp) [inline README.md]: 5, I see the `which include:`

File: README.md (hunk 6 above)
Comment (mrjoelkemp) [inline README.md]: 5. nvm

File: README.md (hunk 7 above)
Comment (kaicataldo) [inline README.md]: Wow, and I'm pretty sure that wasn't an autocorrect. Thanks, brain.

File: src/gulp-paige.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    gutil.log(gutil.colors.red('Required application address missing.'));
+    return;
+  }
+
+  if (!files) {
+    gutil.log(gutil.colors.red('Required test files missing.'));
+    return;
+  }
+
+  if (mochaOptions.grep && (options.suite !== undefined)) {
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: No need for the wrapping parens: http://bit.ly/1Og3d7X

File: src/gulp-paige.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  if (target === 'remote') {
+    if (!webdriver.url) {
+      gutil.log(gutil.colors.red('Required webdriver address missing.'));
+      return;
+    }
+
+    remoteAddress = webdriver.url;
+  }
+
+  files = "'" + files + "'";
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: Template strings would be nice here

File: src/gulp-paige.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+  const webdriverConfig = [
+    webdriver.platform ? '--webdriverPlatform ' + webdriver.platform : undefined,
+    webdriver.browser ? '--webdriverBrowser ' + webdriver.browser : undefined,
+    webdriver.version ? '--webdriverVersion ' + webdriver.version : undefined
+  ].filter(n => n).join(' ');
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: Alex likes doing `.filter(Boolean)` to filter out the elements that are defined

File: src/gulp-paige.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+  if (options.paigeOptions) {
+    paigeOptions = "--additional-config " + options.paigeOptions;
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: Template string

File: src/gulp-paige.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  mochaOptions = [
+    mochaOptions.bail ? '--bail' : undefined,
+    '--slow 30000',
+    '--timeout 20000',
+    '--ui bdd',
+    mochaOptions.grep ? '--grep ' + mochaOptions.grep : undefined,
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: template string

File: src/gulp-paige.js [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    options.suite !== undefined ? '--grep :' + options.suite + ':' : undefined
```
Comment (mrjoelkemp) [inline src/gulp-paige.js]: can't you do `options.suite ? `

File: index.js [hunk 14]
```diff
@@ -27,80 +29,73 @@ function runPaige(config) {
...
-  webdriverVersion = config.webdriverVersion || '';
-
-  gutil.log("Application URI: " + address);
-  gutil.log("WebDriver target: " + webdriverAddress);
-  gutil.log("WebDriver platform: " + webdriverPlatform);
-  gutil.log("WebDriver browser: " + webdriverBrowser);
-  gutil.log("WebDriver version: " + webdriverVersion);
-  gutil.log('');
-  console.log('');
-  console.log('         |\\');
```
Comment (mrjoelkemp) [inline index.js]: Sad to see the horse die

**Comment by mrjoelkemp** [comment]
:+1:

File: src/gulp-paige.js (hunk 10 above)
Comment (kaicataldo) [inline src/gulp-paige.js]: Ah! I like that

**Pull request merged**